src/cryptography/x509.py

```diff
@@ -1344,3 +1344,41 @@ def extensions(self):
         """
         Returns an Extensions object containing a list of Revoked extensions.
         """
+
+
+class CertificateSigningRequestBuilder(object):
+    def __init__(self, subject_name=None, extensions=[]):
+        """
+        Creates an empty X.509 certificate request (v1).
+        """
+        self._subject_name = subject_name
+        self._extensions = extensions[:]
+
+    def subject_name(self, name):
+        """
+        Sets the certificate requestor's distinguished name.
+        """
+        if not isinstance(name, Name):
+            raise TypeError('Expecting x509.Name object.')
+        return CertificateSigningRequestBuilder(name, self._extensions)
```

Comment:
I think I'm leaning towards "yes" on an error -- it's easier to remove the error if we find that it would be useful than it would be to add it in later